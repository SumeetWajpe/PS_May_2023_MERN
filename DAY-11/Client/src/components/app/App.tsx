import React from "react";
import Posts from "../posts/posts.component";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import CourseDetails from "../coursedetails/coursedetails.component";
import PostDetails from "../postdetails/postdetails.component";
import NewCourseWithReactHookForm from "../newcoursewithreacthookform/newcourse.reacthookform";
import Cart from "../cart/cart.component";
import Error from "../error/error.component";
import ListOfCourses from "../listofcourses/listofcourses.component";
import Login from "../login/login.component";
import Dashboard from "../dashboard/dashboard.component";

export default function App() {
  return (
    <>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Login />} />
          <Route path="/dashboard" element={<Dashboard />}>
            <Route index path="" element={<ListOfCourses />} />
            <Route path="coursedetails/:id" element={<CourseDetails />} />
            {/* <Route path="/newcourse" element={<NewCourse />} /> */}
            <Route path="newcourse" element={<NewCourseWithReactHookForm />} />
            <Route path="cart" element={<Cart />} />
            <Route path="posts" element={<Posts />} />
            <Route path="postdetails/:id" element={<PostDetails />} />
          </Route>

          {/* Wild card route handler - 404 */}
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </>
  );
}
