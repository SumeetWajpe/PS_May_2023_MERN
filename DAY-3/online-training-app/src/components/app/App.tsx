import React from "react";
import ListOfCourses from "../listofcourses/listofcourses.component";
import { Message } from "../functional/message.functional";
import { Counter } from "../functional/counter";
import Posts from "../posts/posts.component";
import GetpostById from "../postbyid/getpostbyid";
import { GrandParent } from "../usingcontextapi/usingcontextapi.component";
import { BrowserRouter, Link, Route, Routes } from "react-router-dom";
import CourseDetails from "../coursedetails/coursedetails.component";
import NewCourse from "../newcourse/newcourse.component";

export default function App() {
  return (
    <>
      <BrowserRouter>
        <nav className="navbar navbar-expand-lg bg-body-tertiary">
          <div className="container-fluid">
            <Link className="navbar-brand" to="/">
              Online Training
            </Link>
            <button
              className="navbar-toggler"
              type="button"
              data-bs-toggle="collapse"
              data-bs-target="#navbarSupportedContent"
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation"
            >
              <span className="navbar-toggler-icon"></span>
            </button>
            <div
              className="collapse navbar-collapse"
              id="navbarSupportedContent"
            >
              <ul className="navbar-nav me-auto mb-2 mb-lg-0">
                <li className="nav-item">
                  <Link className="nav-link" aria-current="page" to="/">
                    Courses
                  </Link>
                </li>
                <li className="nav-item">
                  <Link className="nav-link" to="/posts">
                    Posts
                  </Link>
                </li>
                <li className="nav-item">
                  <Link className="nav-link" to="/newcourse">
                    New Course
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </nav>

        <Routes>
          <Route path="/" element={<ListOfCourses />} />
          <Route path="/coursedetails/:id" element={<CourseDetails />} />
          <Route path="/newcourse" element={<NewCourse />} />
          <Route path="/posts" element={<Posts />} />
        </Routes>
      </BrowserRouter>
    </>
  );
}
