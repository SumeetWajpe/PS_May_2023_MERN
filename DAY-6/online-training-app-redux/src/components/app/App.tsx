import React from "react";
import Posts from "../posts/posts.component";
import { BrowserRouter, Link, Route, Routes } from "react-router-dom";
import CourseDetails from "../coursedetails/coursedetails.component";
import PostDetails from "../postdetails/postdetails.component";
import NewCourseWithReactHookForm from "../newcoursewithreacthookform/newcourse.reacthookform";
import Header from "../header/header.component";
import Cart from "../cart/cart.component";
import Error from "../error/error.component";
import ListOfCourses from "../listofcourses/listofcourses.component";

export default function App() {
  return (
    <>
      <BrowserRouter>
        <Header />
        <Routes>
          <Route path="/" element={<ListOfCourses />} />
          <Route path="/coursedetails/:id" element={<CourseDetails />} />
          {/* <Route path="/newcourse" element={<NewCourse />} /> */}
          <Route path="/newcourse" element={<NewCourseWithReactHookForm />} />
          <Route path="/cart" element={<Cart />} />
          <Route path="/posts" element={<Posts />} />
          <Route path="/postdetails/:id" element={<PostDetails />} />

          {/* Wild card route handler - 404 */}
          <Route path="*" element={<Error />} />
        </Routes>
      </BrowserRouter>
    </>
  );
}
