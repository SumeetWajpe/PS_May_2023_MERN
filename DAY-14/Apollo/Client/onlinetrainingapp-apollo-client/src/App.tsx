import React from "react";
import logo from "./logo.svg";
import "./App.css";
import ListOfCourses from "./components/listofcourses.component";

function App() {
  return (
    <div className="App">
      <ListOfCourses />
    </div>
  );
}

export default App;
