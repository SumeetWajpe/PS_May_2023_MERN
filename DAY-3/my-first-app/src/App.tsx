import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { Message } from "./components/Message.component";
import { MessageModel } from "./models/message.model";

class App extends React.Component {
  messageOne: MessageModel = new MessageModel("Hello", "John", "Carter");
  render() {
    return <Message msg={this.messageOne} />;
  }
}

export default App;
