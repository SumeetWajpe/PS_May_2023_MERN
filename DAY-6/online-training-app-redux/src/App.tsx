import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { useSelector } from "react-redux";

function App() {
  const courses = useSelector((store: any) => store.courses);
  return <div className="App">{courses.length}</div>;
}

export default App;
