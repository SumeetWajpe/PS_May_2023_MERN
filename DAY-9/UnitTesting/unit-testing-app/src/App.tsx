import React from "react";
import logo from "./logo.svg";
import "./App.css";

export type CounterState = {
  count: number;
};

export default class App extends React.Component<{},CounterState> {
  state: Readonly<CounterState> = { count: 0 };
  render(): React.ReactNode {
    return (
      <>
        <strong>{this.state.count}</strong>
        <button onClick={() => this.setState({ count: this.state.count + 1 })}>
          ++
        </button>
      </>
    );
  }
}
