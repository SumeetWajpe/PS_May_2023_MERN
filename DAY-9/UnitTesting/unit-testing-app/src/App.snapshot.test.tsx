import renderer from "react-test-renderer";
import App from "./App";
test("renders the app component correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
