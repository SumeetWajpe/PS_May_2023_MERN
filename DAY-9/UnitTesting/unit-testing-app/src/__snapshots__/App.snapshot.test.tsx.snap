// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app component correctly 1`] = `
Array [
  <strong>
    0
  </strong>,
  <button
    onClick={[Function]}
  >
    +++
  </button>,
]
`;
